bugsnag-plugin-android-ndk/src/androidTest/java/com/bugsnag/android/ndk/migrations/EventMigrationV8Tests.kt

```diff
@@ -0,0 +1,224 @@
+package com.bugsnag.android.ndk.migrations
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.fail
+import org.junit.Test
+
+/** Migration v8 added feature flags ⛳️ */
+class EventMigrationV8Tests : EventMigrationTest() {
+
+    @Test
+    /** check notifier and api key, since they aren't included in event JSON */
+    fun testMigrationPayloadInfo() {
+        val infoFile = createTempFile()
+
+        val info = migratePayloadInfo(infoFile.absolutePath)
+
+        assertEquals(
+            mapOf(
+                "apiKey" to "5d1e5fbd39a74caa1200142706a90b20",
+                "notifierName" to "Test Library",
+                "notifierURL" to "https://example.com/test-lib",
+                "notifierVersion" to "2.0.11"
+            ),
+            parseJSON(info)
+        )
+    }
+
+    @Test
+    fun testMigrateEventToLatest() {
+        val eventFile = createTempFile()
+
+        migrateEvent(eventFile.absolutePath)
+        assertNotEquals(0, eventFile.length())
+
+        val output = parseJSON(eventFile)
+
+        assertEquals(
+            "00000000000m0r3.61ee9e6e099d3dd7448f740d395768da6b2df55d5.m4g1c",
+            output["context"]
+        )
+        assertEquals(
+            "a1d34088a096987361ee9e6e099d3dd7448f740d395768da6b2df55d5160f33",
+            output["groupingHash"]
+        )
+        assertEquals("info", output["severity"])
+
+        // app
+        assertEquals(
+            mapOf(
+                "binaryArch" to "mips",
+                "buildUUID" to "1234-9876-adfe",
+                "duration" to 6502L,
+                "durationInForeground" to 12L,
+                "id" to "com.example.PhotoSnapPlus",
+                "inForeground" to true,
+                "isLaunching" to true,
+                "releaseStage" to "リリース",
+                "type" to "red",
+                "version" to "2.0.52",
+                "versionCode" to 57L
+            ),
+            output["app"]
+        )
+
+        // breadcrumbs
+        val crumbs = output["breadcrumbs"]
+        if (crumbs is List<Any?>) {
+            assertEquals(50, crumbs.size)
+            crumbs.forEachIndexed { index, crumb ->
+                assertEquals(
+                    mapOf(
+                        "type" to "state",
+                        "name" to "mission $index",
+                        "timestamp" to "2021-12-08T19:43:50.014Z",
+                        "metaData" to mapOf(
+                            "message" to "Now we know what they mean by 'advanced' tactical training."
+                        )
+                    ),
+                    crumb
+                )
+            }
+        } else {
+            fail("breadcrumbs is not a list of crumb objects?!")
+        }
```

Comment:
```suggestion
        val crumbs = output["breadcrumbs"] as? List<Any?>
            ?: fail("breadcrumbs is not a list of crumb objects?!")
            
        assertEquals(50, crumbs.size)
        crumbs.forEachIndexed { index, crumb ->
            assertEquals(
                mapOf(
                    "type" to "state",
                    "name" to "mission $index",
                    "timestamp" to "2021-12-08T19:43:50.014Z",
                    "metaData" to mapOf(
                        "message" to "Now we know what they mean by 'advanced' tactical training."
                    )
                ),
                crumb
            )
        }
```

Reply:
I tried a few variations here, but all resulted in failed type inference unless there was an explicit type check.